amethyst_tiles/src/map.rs

```diff
@@ -340,22 +326,26 @@ fn to_tile(
     inverse.y = inverse.y.round() * -1.0;
     inverse.z = inverse.z.floor();
 
-    if inverse.x < 0.0 {
-        return None;
-    }
-    if inverse.y < 0.0 {
-        return None;
-    }
-    if inverse.z < 0.0 {
-        return None;
+    if inverse.x < 0.0
+        || inverse.x as u32 > max_dimensions.x
+        || inverse.y < 0.0
+        || inverse.y as u32 > max_dimensions.y
+        || inverse.z < 0.0
+        || inverse.z as u32 > max_dimensions.z
+    {
+        let point_dimensions = Point3::new(inverse.x as i32, inverse.y as i32, inverse.z as i32);
```

Comment:
Think these should be `as u32`

Reply:
The reason they are unsigned is because the tile coordinates you clicked could be negative. This should be passed in the error.

Reply:
oh yes. Does it make sense to pass in whatever type `inverse` is?
Points I'm thinking of:
* is it clear enough for users to understand what the error is, using the raw type?
* it removes the possibility of `i32` overflow

Reply:
Inverse is of type Point3<f32> I believe.

I believe because of the round and floor commands that making it an int is better 

Considering f32 can only store integers accurately up to about 2^23 I wouldn't worry about overflow. 

Changing this to allow for more precision would require fixes outside of the scope of this PR.